chat.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/proto"
+)
+
+func main() {
+	fmt.Println("Type your clientID")
+	reader := bufio.NewReader(os.Stdin)
+	clientId, _ := reader.ReadString('\n')
+	clientId = strings.Replace(clientId, "\n", "", -1)
+
+	opts := &ably.ClientOptions{
+		AuthOptions: ably.AuthOptions{
+			// If you have an Ably account, you can find
+			// your API key at https://www.ably.io/accounts/any/apps/any/app_keys
+			Key: "INSERT_YOUR_API_KEY_HERE",
+		},
+		ClientID: clientId,
+		// NoEcho:   true, // Uncomment to stop messages you send from being sent back
+	}
+
+	fmt.Println("You can now send messages!")
+
+	// Connect to Ably using the API key and ClientID specified above
+	client, err := ably.NewRealtimeClient(opts)
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the Ably Channel with name 'chat'
+	channel := client.Channels.Get("chat")
+
+	// Enter the Presence set of the channel
+	channel.Presence.Enter("")
+
+	getHistory(*channel)
+
+	go subscribeToPresence(*channel)
+
+	go subscribe(*channel)
+
+	// Start the goroutine to allow for publishing messages
+	publishing(*channel)
+}
+
+func subscribeToPresence(channel ably.RealtimeChannel) {
+	sub, err := channel.Presence.Subscribe()
+	if err != nil {
+		panic(err)
```

Comment:
panicking on an error is a bit messy. It would be nicer to return errors, print the error nicely for the user and exit.
I would use an golang.org/x/sync/errgroup for this. Probably no need to use a context in this case.